wamr-compiler/main.c

```diff
@@ -378,6 +483,18 @@ main(int argc, char *argv[])
 
             option.custom_sections_count = len;
         }
+#if BH_HAS_DLFCN
+        else if (!strncmp(argv[0], "--native-lib=", 13)) {
+            if (argv[0][13] == '\0')
+                PRINT_HELP_AND_EXIT();
+            if (native_lib_count >= sizeof(native_lib_list) / sizeof(char *)) {
+                printf("Only allow max native lib number %d\n",
+                       (int)(sizeof(native_lib_list) / sizeof(char *)));
+                return 1;
```

Comment:
Had better `goto fail0` like `PRINT_HELP_AND_EXIT`, which frees `option.custom_sections` if needed. `option.custom_sections` may be allocated in L477 `option.custom_sections = split_string(argv[0] + 23, &len, ",");`

Reply:
OK